### PR TITLE
[processing] keep column names in standard distance matrix (fix #17150)

### PR DESCRIPTION
**Pull request by alexbruy**
## Description
Use values from unique ID field as column names. Fixes https://issues.qgis.org/issues/17150.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by alexbruy** [comment]
@gioman, @nyalldawson is this good for you?

File: python/plugins/processing/algs/qgis/PointDistance.py [hunk 1]
```diff
@@ -225,27 +229,30 @@ def regularMatrix(self, parameters, context, source, inField, target_source, tar
...
+                request = QgsFeatureRequest().setFilterFids(featList).setSubsetOfAttributes([targetIdx])
+                for f in target_source.getFeatures(request):
+                    fields.append(QgsField(f[targetField], QVariant.Double))
+
                 (sink, dest_id) = self.parameterAsSink(parameters, self.OUTPUT, context,
                                                        fields, source.wkbType(), source.sourceCrs())
 
-            data = [inID]
-            for target in target_source.getFeatures(QgsFeatureRequest().setSubsetOfAttributes([]).setFilterFids(featList).setDestinationCrs(source.sourceCrs())):
+            data = [str(inFeat[inField])]
```
Comment (m-kuhn) [inline python/plugins/processing/algs/qgis/PointDistance.py]: Is the cast to `str` required?

Comment (alexbruy) [reply python/plugins/processing/algs/qgis/PointDistance.py]: Yes, because unique ID field can be numeric

Comment (m-kuhn) [reply python/plugins/processing/algs/qgis/PointDistance.py]: My question was rather, why is it forced to be non-numeric in the output. Sorry if I missed something obvious.

Comment (alexbruy) [reply python/plugins/processing/algs/qgis/PointDistance.py]: Ah sorry, my fault. You are right, this cast is not needed here. Fixed

File: python/plugins/processing/algs/gdal/Buffer.py [hunk 2]
```diff
@@ -115,6 +115,10 @@ def getConsoleCommands(self, parameters, context, feedback):
             if f.name() == geometry:
                 continue
             other_fields.append(f.name())
+        if len(other_fields) > 0:
```
Comment (m-kuhn) [inline python/plugins/processing/algs/gdal/Buffer.py]: Should this file be in here?
But I guess that should that read `if other_fields:`

PEP 8:

> For sequences, (strings, lists, tuples), use the fact that empty sequences are false.

**Comment by nyalldawson** [comment]
Is this pr correct? The commit looks unrelated

**Comment by alexbruy** [comment]
Should be fixed now

**Review by m-kuhn — APPROVED** [review]
Looks good to me

**Comment by nyalldawson** [comment]
Looks good to me - is there no unit tests for this alg?

**Comment by alexbruy** [comment]
@nyalldawson added tests for regular and summary distance matrix. Tried also to add test for linear matrix, but it always failed for me (probably because of features order, as result looks correct).

**Comment by nyalldawson** [comment]
>  always failed for me (probably because of features order, as result looks correct).

There's a way to specify the feature order for the tests - see e.g https://github.com/qgis/QGIS/blob/master/python/plugins/processing/tests/testdata/qgis_algorithm_tests.yaml#L1441

Basically you specifiy the primary key ('pk') as either a single field or ordered list of fields, and the comparison will be done after having sorted the expected/obtained layers by those key fields.

**Comment by alexbruy** [comment]
Thanks! Tests for all 3 matrix types now enabled

**Pull request merged**